share/ci/git_rebase.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+# rebase the PR to the latest version of the destination branch
+
+cd $CI_PROJECT_DIR
+
+github_group_repo="ComputationalRadiationPhysics/picongpu"
+
+pr_id=$(echo "$CI_BUILD_REF_NAME" | cut -d"/" -f1 | cut -d"-" -f2)
+# used a token without any rights from psychocoderHPC to avoid API query limitations
+echo "curl -w -u psychocoderHPC:$GITHUB_TOKEN -X GET https://api.github.com/repos/${github_group_repo}/pulls/${pr_id} 2>/dev/null"
+curl_data=$(curl -u psychocoderHPC:$GITHUB_TOKEN -X GET https://api.github.com/repos/${github_group_repo}/pulls/${pr_id} 2>/dev/null)
+echo "curl data"
+echo "$curl_data"
+# get the destination branch
+target_branch=$(echo "$curl_data" | python3 -c 'import json,sys;obj=json.loads(sys.stdin.read());print(obj["base"]["ref"])')
+echo "target_branch=${target_branch}"
+
+mainline_exists=$(git remote -v | cut -f1 | grep mainline -q && echo 0 || echo 1)
+# avoid that we add the remote reposetory twice if gitlab already cached this operation
```

Comment:
Typos, can be re-written as `# avoid adding the remote repository twice if gitlab already cached this operation`